src/admin-data-import/components/AdminDataImportPage/ImportSummaryModal.jsx

```diff
@@ -0,0 +1,102 @@
+                                <p className='error'>
+                                    {formatMessage('admin.dataImport.modal.failedEntries')}: <strong>{results[currentTab].failedEntriesCount}</strong>
+                                </p>
+                            </div>
+
+                            {(results[currentTab].failedEntriesCount > 0 && results[currentTab].errors?.length) && (
+                                <table className='failed-entries-table'>
+                                    <thead>
+                                        <tr>
+                                            {results[currentTab].errors.length > 0 &&
+                                                Object.keys(results[currentTab].errors[0]).map((key, index) => (
+                                                    key !== 'errors' &&
+                                                    <th key={index}>{formatMessage(`${IMPORT_SUMMARY_HEADER_TRANSLATION_PREFIX}.${key}`)}</th>
+                                                ))}
+                                            <th>{formatMessage('admin.dataImport.modal.errors')}</th>
+                                        </tr>
+                                    </thead>
+                                    <tbody>
+                                        {results[currentTab].errors.map((error, rowIndex) => (
+                                            <tr key={rowIndex}>
+                                                {Object.keys(error).map((key, colIndex) => (
+                                                    key !== "errors" && <td key={colIndex}>{error[key]}</td>
+                                                ))}
+                                                <td>
+                                                    <ul>
+                                                        {error.errors.map((errMsg, i) => (
+                                                            <li key={i}>{formatMessage(errMsg)}</li>
+                                                        ))}
+                                                    </ul>
+                                                </td>
+                                            </tr>
+                                        ))}
+                                    </tbody>
+                                </table>
+                            )}
+                        </div>
+
+                        <div
+                            className='react-modal-footer'
+                            style={{
+                                flexDirection: 'row-reverse'
+                            }}>
+                            <button className='btn primary' onClick={onClose}>
+                                {formatMessage('admin.dataImport.modal.confirm')}
+                            </button>
+                        </div>
+                    </>
+                }
+
+
+            </Modal>
+        </>
+
+    )
+}
```

Comment:
Few suggestions to the whole ImportSummaryModal component:
1. Add an early return for empty results: `   if (!results || !results.length) {
     return null;
   }`
2. Extract the current result to a variable to improve readability: `const currentResult = results[currentTab];
`.
3. Move the complex error condition out of JSX: `const hasErrors = 
  currentResult.failedEntriesCount > 0 && 
  currentResult.errors?.length > 0;`

Reply:
Moreover, I noticed you're using an index as a key. This is acceptable only if the table's order remains fixed, ensuring items are never deleted or added. Take a look at this to see if it meets the conditions.